src/Pim/Bundle/CatalogBundle/Doctrine/Common/Saver/ProductSaver.php

```diff
@@ -115,4 +116,49 @@ protected function validateProduct($product)
             );
         }
     }
+
+    /**
+     * Drops the current completenesses from the database for a list of products.
+     *
+     * @param ProductInterface[] $products
+     */
+    protected function dropCompletenesses(array $products)
+    {
+        $completenessIDs = [];
+        foreach ($products as $product) {
+            $completenesses = $product->getCompletenesses();
+            foreach ($completenesses as $completeness) {
+                $completenessIDs[] = $completeness->getId();
+            }
+        }
+
+        $stmt = $this->entityManager->getConnection()->executeQuery(
```

Comment:
we should use the method delete from the doctrine connection instead of hard coded query. Perhaps, you should call an external service to do that :)